### PR TITLE
Outlaws negative molar counts

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

You ready for this one?
Atmos + space can sometimes create negative molar counts.
I first noticed this issue around 3 months ago with this bug #48533, which this should fix by the way. 

I initially thought this was caused by the sm going negative with one of it's internal values, but that proved not to be the case. In an attempt to fix it I pr'd #49081, which in actuality only neutered the effects somewhat.

Now to go over what was added here.
You'll notice I've added a couple more lines to the QUANTIZE comment. They attempt to explain briefly what I'll go over here.

The bug that I believe is the root of this issue is to do with round().
Round(0.5, 1) will output 1, rather then 0. Basically round can round up.

My current working theory is this is fucking with molar counts by subtracting more then was originally present on tiles, which I know is happening. 

I, along with duncathen, have tried a few different replacements for QUANTIZE(), all of which broke shit.

Flooring it appears to cause issues with high molar counts, and appears cause a loss in accuracy earlier then expected.
If statements are not clear enough, and do not appear to fix the underlying bug.
Nor does mixing the two.
Life is pain.

What I'm pr'ing here is a fix I've had in my back pocket for a while. It's been tested against ordinary atmos behavior, so fusion spread, air tanks, sm, ect. it does not appear to cause anomalous behavior.

It also absolutely disallows negative molar counts on turfs, which appears to be intended.

We do this by making 2 changes. 
The first is to make garbage collect only care about the actual molar count, rather then also caring about the archived one. This is to fix the garbage collects in share() not cleaning up properly due to a turfs actual molar count being negative, but it's archived value being positive.

The second is to remove the optimization in share() that assumed that if a gas was in both mixes it didn't need to be gc'd.

Both of these are necessary to patch over this issue. Neither should cause any unexpected behavior.
For anyone who comes after me. Keep in mind that quantize needs to allow for the same reaction to negative molar counts, and round() does not behave in that manner. For testing purposes fix this with a ternary statement such as `round(((input > 0) ? 1 :  -1) * input, ACCURACY) * ((input > 0) ? 1 :  -1)`

I have yet to describe how to precisely reproduce the issue with the sm, this is to prevent issues if this does not get merged, or has unforeseen issues on merge.

To check if your method works, loop through all the turfs being processed at the start of airss, and check if they have a negative molar count.

## Why It's Good For The Game

Fixes my insomnia
## Changelog
:cl: LemonInTheDark, with thanks to: Hugbug for helping me reproduce and test untold times, Dennok for brainstorming the round() issue, duncathan for guiding me through QUANTIZE changes, Ghilker for helping me test the more arcane atmos stuff, as334 for assisting with debugging. Thank you all.
fix: Outlaws negative molar counts from the atmos system.
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 1]
```diff
@@ -5,8 +5,12 @@ What are the archived variables for?
 */
 #define MINIMUM_HEAT_CAPACITY	0.0003
 #define MINIMUM_MOLE_COUNT		0.01
-#define QUANTIZE(variable)		(round(variable,0.0000001))/*I feel the need to document what happens here. Basically this is used to catch most rounding errors, however it's previous value made it so that
-															once gases got hot enough, most procedures wouldnt occur due to the fact that the mole counts would get rounded away. Thus, we lowered it a few orders of magnititude */
+#define ACCURACY 1E-7
```
Comment (duncathan) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm:8]: do you wanna rename this as we discussed in DMs? maybe `MOLAR_ACCURACY`

**Review by duncathan — CHANGES_REQUESTED** [review]

**Review by duncathan — COMMENTED** [review]
we worked on this pretty rigorously to try and come up with a solution. I'm confident this is the needed change for now

**Review by duncathan — APPROVED** [review]

**Comment by Ghilker** [comment]
finally some good atmos shit

**Comment by LemonInTheDark** [comment]
I know right?

**Comment by LemonInTheDark** [comment]
Functionally reverted by https://github.com/tgstation/tgstation/pull/63785, the core problem was not rounding, and was instead out of order shares. it's fixed now, and god I'm happy

**Pull request merged**